product-store-web/src/app/general/services/products.service.ts

```diff
@@ -41,6 +42,31 @@ export class ProductsService {
       });
   }
 
+  public getProduct(filter: ProductFilter): Observable<Product> {
+    const queryFilter = `filter: { id: "${filter.id}" }`;
```

Comment:
hand in key + value via filter

Reply:
Yeah... I thought that it's maybe better to specify exactly which possibilities are possible as filtering. If we would do it like this, we would force the person to look at the filtering possibilities in the BE. Like this, it's fixed what can be filtered.